### DISCUSSION TITLE
Update Error! On CheckForUpdates if App Sandbox=YES In Release env

### DISCUSSION THREAD

**Discussion by iOSBoy** [sparkle-project/Sparkle#2540] [Q&A] [answered]
### Description of the problem
CheckForUpdate fails with "Update Error! An error occurred while launching the installer. Please try again later.
It's work very well on debug env when i use xcode launch my app ,but where i use xcode archive a release env dmg  then check update by update menu,An error occurred when the dmg finished download.

![image](https://github.com/sparkle-project/Sparkle/assets/11255978/dbdaa2d0-45a4-4cb2-b419-e474bbcf99fe)

### Do you use Sandboxing in your app?

YES

### Version of `Sparkle.framework` in the latest version of your app
2.6.0
### Version of `Sparkle.framework` in the old version of app that your users have (or N/A)
N/A
### Sparkle's output from Console.app
```
Sandbox: uhomeaaa(19041) deny(1) mach-lookup com.xxx.aaa-spks
Violation:       deny(1) mach-lookup com.xxx.aaa-spks
Process:         uhomeaaa [19041]
Path:            /Applications/uhomeaaa.app/Contents/MacOS/uhomeaaa
Load Address:    0x1049b0000
Identifier:       com.xxx.aaa
Version:         27 (2.5.14)
Code Type:       arm64 (Native)
Parent Process:  uhomes pro [1]
Responsible:     /Applications/uhomeaaa.app/Contents/MacOS/uhomeaaa
User ID:         501

Date/Time:       2024-04-09 17:35:00.142 GMT+8
OS Version:      macOS 14.3.1 (23D60)
Release Type:    User
Report Version:  8

MetaData: {"checker":"launchd","summary":"deny(1) mach-lookup com.xxx.aaa-spks","policy-description":"Sandbox","profile-flags":0,"platform_binary":"no","mach_namespace":1,"binary-in-trust-cache":false,"uid":501,"responsible-process-team-id":"7794444B9HL","primary-filter":"global-name","operation":"mach-lookup","flags":5,"platform-binary":false,"global-name":"com.xxx.aaa-spks","build":"macOS 14.3.1 (23D60)","apple-internal":false,"process-path":"\/Applications\/uhomeaaa.app\/Contents\/MacOS\/uhomeaaa","platform-policy":false,"hardware":"J314s","pid":19041,"process_path":["Applications","uhomeaaa.app","Contents","MacOS","uhomeaaa"],"checker-pid":1,"team-id":"7794444B9HL","normalized_target":["com.xxx.aaa-spks"],"primary-filter-value":"com.xxx.aaa-spks","action":"deny","sandbox_checker":"launchd","process":"uhomeaaa","signing-id":"com.xxx.aaa","errno":1,"profile-in-collection":false,"target":"com.xxx.aaa-spks","responsible-process-signing-id":"com.xxx.aaa","responsible-process-path":"\/Applications\/uhomeaaa.app\/Contents\/MacOS\/uhomeaaa","container":"\/Users\/tao\/Library\/Containers\/F375DEF2-F58A-468F-B6F9-FB613749F192\/Data","release-type":"User"}

=============
Error: An error occurred while running the updater. Please try again later. The remote port connection was invalidated from the updater. For additional details, please check Console logs for Autoupdate. If your application is sandboxed, please also ensure Installer Connection & Status entitlements are correctly set up: https://sparkle-project.org/documentation/sandboxing/ (URL (null))

```
### Steps to reproduce the behavior

[The more information provided and pasted verbatim, the easier it will be to diagnose an issue. If you can provide the affected application/binary and XML feed to reproduce an issue, share them]
<img width="844" alt="image" src="https://github.com/sparkle-project/Sparkle/assets/11255978/5058757c-2af8-4634-8d26-56a78e362b6e">

<img width="907" alt="image" src="https://github.com/sparkle-project/Sparkle/assets/11255978/e5a4a4bc-8c7f-483c-98ad-06393406bb4d">

<img width="982" alt="image" src="https://github.com/sparkle-project/Sparkle/assets/11255978/2d3f6ae3-cbc8-4684-826f-6362cb557009">

**Comment by zorgiepoo** [comment, marked as answer]
This is very likely a sandbox configuration issue and a difference between your debug / release-archived builds.

As you showed you are getting `deny(1) mach-lookup com.xxx.aaa-spks`. This may mean in your release build the entitlements for looking up `com.xxx.aaa-spks` are stripped. You can try `codesign -d --entitlements - --xml /Applications/uhomeaaa.app/Contents/MacOS/uhomeaaa` on your archived build to view its entitlements and see if it's still present.

The guide mentions this. I would check if you have any custom (re-)signing anywhere in your project. Just using Xcode's Archiving should preserve entitlements correctly.

> Due to different code signing requirements, please do not add --deep to OTHER_CODE_SIGN_FLAGS or from custom build scripts when signing your application. This is a common source of Sandboxing errors.

**Reply by iOSBoy** [reply]
Hi @zorgiepoo  Thank you for your detail rely ,I have recheck my release-archived progress it's has re-sign step, I had remove this step and then passed the test, thanks again.
